client/components/inline-support-link/use-support-doc-data.ts

```diff
@@ -0,0 +1,82 @@
+import { localizeUrl } from '@automattic/i18n-utils';
+import { dispatch } from '@wordpress/data';
+import { useEffect, useState } from 'react';
+import type { ContextLinks, SupportDocData } from './types';
+import type { HelpCenterDispatch } from '@automattic/data-stores';
+
+const HELP_CENTER_STORE = 'automattic/help-center';
+
+const loadHelpCenterDispatch = async () => {
+	// Load `@automattic/data-stores` asynchronously to avoid including it in the main bundle and reduce initial load size.
+	if ( ! dispatch( HELP_CENTER_STORE ) ) {
+		const { HelpCenter: HelpCenterStore } = await import(
+			/* webpackChunkName: "async-load-automattic-data-stores" */ '@automattic/data-stores'
+		);
+		HelpCenterStore.register();
+	}
+
+	return dispatch( HELP_CENTER_STORE ) as HelpCenterDispatch[ 'dispatch' ];
+};
+
+const useSupportDocData = ( {
+	supportPostId = 0,
+	supportLink = '',
+	supportContext,
+}: {
+	supportPostId?: number;
+	supportLink?: string;
+	supportContext?: string;
+} ): {
+	supportDocData: SupportDocData | null;
+	openSupportDoc: () => Promise< void > | void;
+} => {
+	const [ supportDocData, setSupportDocData ] = useState< SupportDocData >( {
+		link: supportPostId && supportLink ? localizeUrl( supportLink ) : supportLink,
```

Comment:
Question: Why does the link need to check for supportPostId?

Reply:
I followed the existing behavior but maybe we can always wrap `supportLink ` with `localizeUrl` if it's not empty 🙂